### ISSUE TITLE
[Need help] Instrumented code only sends one message, no coverage being dumped

### ISSUE THREAD

**Issue by dhrimov** [cqse/teamscale-javascript-profiler#111]
Hello!

I am trying to use the tool on an app (web app) to collect coverage during automated UI test runs.

Well, so far:
- I've run instrumenter successfully
- I can run my app
- I started collector
- Once app launches I can see logs:
```
Starting coverage forwarding worker.
Received coverage mapping information for "dd26609bf0e13e1e77ee3dd93d50ea17774a92b5".
Connection to Coverage Collector established.
```
- If I go to network tab, find the websocket connection and go to messages - I can see one message sent. source map details, seems like.

So here I am. As I click through the app, I do not see any other websocket messages sent.
Collector does not log anything upon the initial message receipt. (I can see a path inside, should I run collector from a specific place? I actually tried it.) still, would be nice to have a lot line saying that initial connection ok, code is available, everything is visible etc.
I assume because web app does not send any new ws messages, I get 0 lines coverage dumps on the way.

Happy to provide more info, just not sure what else is relevant.

Thank you!

**Comment by stahlbauer** [comment]
Hi. 

Thanks for reaching out. 

Have you specified any include/exclude patterns? What if you do not use such includes/excludes at all? Have you looked into the instrumented bundles: Have coverage statements been added at the relevant lines/statements/expressions?

**Comment by dhrimov** [comment]
Hi.

1) Yes, I have include pattern. Tried several different options. Any of these seemed to work.
2) Did not try without these because of the project structure
3) I have not. But I have looked at the size. Size jumped to 90MB, so I assume code was instrumented. Will double check and get back to you with this.

**Comment by stahlbauer** [comment]
Note that with the now release version `0.1.0-beta.1`, the overall performance should increase considerably. Furthermore, the bundle sizes should get smaller.

**Comment by dhrimov** [comment]
Hello!

So, I tried `0.1.0-beta.2` and the instrumentation speed is so much better, thank you!
But I am still with the same error.

So, bundle is 60MB (down from 90MB).
On top of it I can see a mapping: `const _$o3wj733 = "app/page/path/file.js";`

If I search for `o3wj733` there is only 1 match and it is a const declaration.

No websocket message is being sent. No coverage is being dumped.

I use `include-origin` in the instrumenter command.

**Comment by stahlbauer** [comment]
Specifying patterns is often not as straigtforward as it seems. Please make sure to quote the pattern in a way that avoids pattern expansion as done by many command line interfaces, for example, by quoting the patterns using single quotes (`'`), such as `--include-origin '**/*.js'`. We expect a GLOB pattern---see https://github.com/cqse/teamscale-javascript-profiler/blob/master/packages/teamscale-javascript-instrumenter/test/unit/Task.test.ts for examples. You should test if the path that is stored in `_$o3wj733` is matched by the pattern you specified.

**Comment by stahlbauer** [comment]
@dhrimov Are you still facing the problem?

**Comment by dhrimov** [comment]
hi. @stahlbauer I got priorities switched, so I have not had a chance to try your last comment, sorry. will get back to you as soon as I can.

**Comment by stahlbauer** [comment]
We just released version `0.1.0-beta.4` with some fixes and enhancements in context of origin file matching, including the statistics outlined above. Relative paths starting with `../` were not matched properly if not specified explicitly.

**Comment by stahlbauer** [comment]
@dhrimov Closing this issue for now. Feel free to re-open this or another issue. 

We would also be interested to hear about your experience using the profiler and your use cases. To ensure quick response from our side, you can also use our support address support@teamscale.com. In case you would be interested in Test Gap Analysis for your organization, our Test Intelligence Team would be happy to answer questions and demonstrate some of the functionality we provide in context of our coverage profilers and Teamscale, our Software Intelligence Platform.
